decode_test.go

```diff
@@ -758,22 +758,23 @@ type DoublePtr struct {
 
 var unmarshalTests = []unmarshalTest{
 	// basic types
-	{in: `true`, ptr: new(bool), out: true},                                                                                                                       // 0
-	{in: `1`, ptr: new(int), out: 1},                                                                                                                              // 1
-	{in: `1.2`, ptr: new(float64), out: 1.2},                                                                                                                      // 2
-	{in: `-5`, ptr: new(int16), out: int16(-5)},                                                                                                                   // 3
-	{in: `2`, ptr: new(json.Number), out: json.Number("2"), useNumber: true},                                                                                      // 4
-	{in: `2`, ptr: new(json.Number), out: json.Number("2")},                                                                                                       // 5
-	{in: `2`, ptr: new(interface{}), out: float64(2.0)},                                                                                                           // 6
-	{in: `2`, ptr: new(interface{}), out: json.Number("2"), useNumber: true},                                                                                      // 7
-	{in: `"a\u1234"`, ptr: new(string), out: "a\u1234"},                                                                                                           // 8
-	{in: `"http:\/\/"`, ptr: new(string), out: "http://"},                                                                                                         // 9
-	{in: `"g-clef: \uD834\uDD1E"`, ptr: new(string), out: "g-clef: \U0001D11E"},                                                                                   // 10
-	{in: `"invalid: \uD834x\uDD1E"`, ptr: new(string), out: "invalid: \uFFFDx\uFFFD"},                                                                             // 11
-	{in: "null", ptr: new(interface{}), out: nil},                                                                                                                 // 12
-	{in: `{"X": [1,2,3], "Y": 4}`, ptr: new(T), out: T{Y: 4}, err: &json.UnmarshalTypeError{"array", reflect.TypeOf(""), 7, "T", "X"}},                            // 13
-	{in: `{"X": 23}`, ptr: new(T), out: T{}, err: &json.UnmarshalTypeError{"number", reflect.TypeOf(""), 8, "T", "X"}}, {in: `{"x": 1}`, ptr: new(tx), out: tx{}}, // 14
-	{in: `{"x": 1}`, ptr: new(tx), out: tx{}}, // 15, 16
+	{in: `true`, ptr: new(bool), out: true},                                                                                            // 0
```

Comment:
This test case is based on the encoding/json one. It's not formatted to reduce diffs, so don't make unnecessary diffs

Reply:
Understandable.
Should I remove this changes?

Reply:
Yes, please 🙏

Reply:
Done.